master/internal/core.go

```diff
@@ -1053,6 +1053,16 @@ func (m *Master) Run(ctx context.Context) error {
 
 	if m.config.Observability.EnablePrometheus {
 		p := prometheus.NewPrometheus("echo", nil)
+		// Group and obscure URLs returning 400 or 500 errors outside of /api/v1 and /det
+		p.RequestCounterURLLabelMappingFunc = func(c echo.Context) string {
+			if strings.HasPrefix(c.Path(), "/det/") || strings.HasPrefix(c.Path(), "/api/v1/") {
```

Comment:
Is there a DRY mechanism to get these prefixes from some other "known listening prefixes" list?

Either way, this only partially mitigates the problem; a motivated attacker could still just hit /det/nosuchpage1, for example, and cause the same cardinality explosion.  I'd be inclined to simply group all response codes regardless of prefix in the metrics. Diagnosing what individual URIs are causing an increase in error responses is more of a job for logs than metrics.

Reply:
I guess I could see a coarse grouping (api v/s ui v/s unknown, perhaps).

Reply:
/det/* and /api/v1/* paths are already grouped to /det/* and /api/v1/* "buckets" so this wouldn't really happen. i agree though that ideally we could have some base paths that are sensible "buckets", but currently /det/ and /api/v1/ resolve to the same echo handlers and there's no easy way to come up with a base bucket path.